pulp_rpm/app/advisory.py

```diff
@@ -190,6 +194,25 @@ def resolve_advisory_conflict(previous_advisory, added_advisory):
     return to_add, to_remove, to_exclude
 
 
+def _copy_update_collections_for(the_collections):
+    """
+    Deep-copy each UpdateCollection in the_collections, and its UpdateCollectionPackages.
+    """
+    new_collections = []
+    for collection in the_collections:
```

Comment:
```
for collection in the_collections.select_related('packages').iterator()
```

Reply:
(this addresses the N+1 problem)

Reply:
I need to do work on/with the collection itself first, before I start working on the packages.

Reply:
select related doesn't change what type of objects you get back, it just joins them for you so that when you access ```collection.packages.all()``` it doesn't need to perform another query in a loop.

unless I'm misunderstanding what your concern is

Reply:
Ahhh, ok ok - no I was just misunderstanding what you were proposing, got it thanks!

Reply:
I had the exact same confusion the first 10 times I saw it, so you're in good company :)

Reply:
select_related() only works for one-to-one relations, and packages is many-to-many, alas.

Reply:
select_related() can only be used for 1-to-1 relations, and 'packages' is many-to-many. prefetch_related() does what we want tho; change will be in next push